### PR TITLE
Generate development DocC on PR merges

### PR DESCRIPTION
**Pull request by OdNairy**
## Pull request checklist:
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. Put tests in correct [Test Plan](https://github.com/mapbox/mapbox-maps-ios/tree/main/Tests/TestPlans) (Unit, Integration, All)
   - [ ] If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: scripts/release/create-docs-prs.sh [hunk 1]
```diff
@@ -25,11 +25,16 @@ main() {
...
     [ -n "$VERSION" ]
     brew_install_if_needed jq
-    brew_install_if_needed gh
 
     step "Update mapbox/maps-ios@publisher-staging"
     maps_ios_upload_docs
 
+    # if STAGING_ONLY is set to true, then exit
+
+    # Exit is should be staging only
```
Comment (pjleonard37) [inline scripts/release/create-docs-prs.sh:34]:
```suggestion
```
I recommend removing this as the above comment says the same

**Review by pjleonard37 — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -1353,6 +1369,48 @@ jobs:
...
+      - store_artifacts:
+          path: MapboxMaps.doccarchive.zip
+      - persist_to_workspace:
+          root: .
+          paths:
+            - latest
+
+  upload-docc:
+    executor: xcode-latest
+    steps:
```
[the remark this answers is not in the document]
Comment (OdNairy) [reply .circleci/config.yml:1404]: Done

File: scripts/release/create-docs-prs.sh [hunk 3]
```diff
@@ -66,9 +72,20 @@ maps_ios_upload_docs() {
     info "Push"
     git push --force --quiet
 
-    VERSION_BRANCH_NAME="docs/$VERSION"
-    git checkout -b "$VERSION_BRANCH_NAME"
-    git push --set-upstream origin "$VERSION_BRANCH_NAME" --force --quiet
+    if [ "$STAGING_ONLY" = true ]; then
+        git push --force
```
[the remark this answers is not in the document]
Comment (OdNairy) [reply scripts/release/create-docs-prs.sh:74]: DocC push event payload is too huge for AWS functions. We have to make a separate trigger to start deployment

File: scripts/release/create-docs-prs.sh (hunk 1 above)
Comment (OdNairy) [reply scripts/release/create-docs-prs.sh]: Done

File: .circleci/config.yml [hunk 4]
```diff
@@ -1353,6 +1369,48 @@ jobs:
...
+      - checkout
+      - configure-environment:
+          skip-dependencies: true
+      - attach_workspace:
+          at: .
+      - run:
+          name: Create documentation pull requests
+          command: |
+            scripts/release/create-docs-prs.sh -sep latest
```
[the remark this answers is not in the document]
Comment (OdNairy) [reply .circleci/config.yml:1413]: Sorry, don't see how is that related

**Pull request merged**